module-system/sov-state/src/internal_cache.rs

```diff
@@ -50,9 +50,7 @@ impl StorageInternalCache {
         let cache_value = self.get_value_from_cache(cache_key.clone());
 
         match cache_value {
-            cache::ValueExists::Yes(cache_value_exists) => {
-                cache_value_exists.map(StorageValue::new_from_cache_value)
-            }
+            cache::ValueExists::Yes(cache_value_exists) => cache_value_exists.map(Into::into),
```

Comment:
Unrelated to this PR, but it seems the API of `ValueExists` could improve as it really looks like a boolean. It could, for instance, deref to an actual boolean so combinators are available.

Reply:
We can open an issue for that